### DISCUSSION TITLE
Build AssImp with MinGW64 in windows

### DISCUSSION THREAD

**Discussion by pparunkc** [assimp/assimp#5633] [Q&A] [answered]
Hi guys,
I am very new to the AssImp and building with CMake. I have been able to build for Visual studio quite well, but unable to do it for QT with MinGW. I have been reading docs and trying to build for more than week, but all my efforts ended up in failures. 
Is it possible to guide me the process of building AssImp with MinGW as none of the resources in the internet has helped. 

OS: Windows 11 x64
Qt 6.6, MinGW 1120 and CMAKE 3.9

Thank you

**Comment by kimkulling** [comment]
Could you add some compiler errors, please?

**Comment by ThatOSDev** [comment, marked as answer]
I use MinGWx64 On windows 10.

Clone Assimp to your desktop.

Then open a command prompt, and navigate into the assimp folder.

While in that folder, type the following, yes even the quotes are included ...

```
cmake CMakeLists.txt -D BUILD_SHARED_LIBS=OFF -D CMAKE_C_COMPILER="C:/path/to/your/gcc/bin/gcc.exe" -D CMAKE_CXX_COMPILER="C:/path/to/your/gcc/bin/g++.exe" -G "MinGW Makefiles"
```
It will create some make files. Then just type make at the prompt.
```
make
```
That's all there is to it.
You will see a lib folder added and it will have libassimp.a inside that folder. Copy the LIB and the INCLUDE to your project.

Point the paths to lib and include and add ...
```
-lgdi32 -lassimp -lstdc++ -lz
```
 as libraries for your project. The -LZ is so that assimp will find the zip library. The -LSTDC++ is so that you can access the C++ functions within the assimp library. Make sure you add those libraries IN THE ORDER I POSTED.

Enjoy

**Comment by ThatOSDev** [comment]
MAKE SURE you put in the correct paths to your GCC and G++ exe files in that line I posted above.

Also, if you are adding another library, like QT, then you would insert it into the libraries includes like this ...

```
-lgdi32 -lqt -lassimp -lstdc++ -lz
```

Notice where I add QT. Assimp and others behind it MUST be last. Order matters.

Cheers

**Comment by ThatOSDev** [comment]
Another thing I forgot to add, I use C, not C++.

So if you are adding assimp to your project, you might not need -lstdc++ as an addition to your libraries for your project. I don't know for sure, since I do not code in C++.

Cheers

**Comment by pparunkc** [comment]
Hi,

Thank you for the response. I am able to build libassimp.a with CMake.

Thank you,


[cid:passionaperture18pxclear_510941e8-603d-4590-b8d5-3e03a689ddc8.png]
Arun Krishnan Chythanya
Pipeline TD

33-34 Rathbone Place
London W1T 1JN
+44 (0)20 7323 9933

www.passion-pictures.com<http://www.passion-pictures.com/>

[cid:passionwordmark18pxclear_39cdd97e-c668-46b1-8224-7b305b41ffc0.png]
[cid:supermanbannerv2_c6edaa53-cd2e-4f1a-8b6e-cc389d52b083.png]

Registered Address: Passion Pictures Ltd, 33-34 Rathbone Place, London W1T 1JN.

Registration number: 2177709. Registered in England & Wales


Disclaimer

This message (and any associated files) is intended only for the
use of the individual or entity to which it is addressed and may
contain information that is confidential, subject to copyright or
constitutes a trade secret. If you are not the intended recipient
you are hereby notified that any dissemination, copying or
distribution of this message, or files associated with this message,
is strictly prohibited. If you have received this message in error,
please notify us immediately by replying to the message and deleting
it from your computer. Messages sent to and from us may be monitored.

Internet communications cannot be guaranteed to be secure or error-free
as information could be intercepted, corrupted, lost, destroyed, arrive
late or incomplete, or contain viruses. Therefore, we do not accept
responsibility for any errors or omissions that are present in this
message, or any attachment, that have arisen as a result of e-mail
transmission. If verification is required, please request a hard-copy
version. Any views or opinions presented are solely those of the author
and do not necessarily represent those of the company.
________________________________

**Comment by ThatOSDev** [comment]
All good, glad you got it solved. If you could mark this as answered. Thanks.

**Comment by pparunkc** [comment]
Sure.

Thanks,


[cid:passionaperture18pxclear_510941e8-603d-4590-b8d5-3e03a689ddc8.png]
Arun Krishnan Chythanya
Pipeline TD

33-34 Rathbone Place
London W1T 1JN
+44 (0)20 7323 9933

www.passion-pictures.com<http://www.passion-pictures.com/>

[cid:passionwordmark18pxclear_39cdd97e-c668-46b1-8224-7b305b41ffc0.png]
[cid:supermanbannerv2_c6edaa53-cd2e-4f1a-8b6e-cc389d52b083.png]

Registered Address: Passion Pictures Ltd, 33-34 Rathbone Place, London W1T 1JN.

Registration number: 2177709. Registered in England & Wales


Disclaimer

This message (and any associated files) is intended only for the
use of the individual or entity to which it is addressed and may
contain information that is confidential, subject to copyright or
constitutes a trade secret. If you are not the intended recipient
you are hereby notified that any dissemination, copying or
distribution of this message, or files associated with this message,
is strictly prohibited. If you have received this message in error,
please notify us immediately by replying to the message and deleting
it from your computer. Messages sent to and from us may be monitored.

Internet communications cannot be guaranteed to be secure or error-free
as information could be intercepted, corrupted, lost, destroyed, arrive
late or incomplete, or contain viruses. Therefore, we do not accept
responsibility for any errors or omissions that are present in this
message, or any attachment, that have arisen as a result of e-mail
transmission. If verification is required, please request a hard-copy
version. Any views or opinions presented are solely those of the author
and do not necessarily represent those of the company.
________________________________
